sdk/go/x/auto/example_test.go

```diff
@@ -1073,3 +1075,98 @@ func ExampleStack_SetAllConfig() {
 	// set all config in map
 	_ = stack.SetAllConfig(ctx, cfg)
 }
+
+func ExampleStack_Cancel() {
+	ctx := context.Background()
+	stackName := FullyQualifiedStackName("org", "project", "stack")
+	stack, _ := SelectStackLocalSource(ctx, stackName, filepath.Join(".", "program"))
+	// attempt to cancel the in progress operation
+	// note that this operation is _very dangerous_, and may leave the stack in an inconsistent state.
+	_ = stack.Cancel(ctx)
+}
+
+func ExampleStack_Export() {
+	ctx := context.Background()
+	stackName := FullyQualifiedStackName("org", "project", "stack")
+	stack, _ := SelectStackLocalSource(ctx, stackName, filepath.Join(".", "program"))
+	dep, _ := stack.Export(ctx)
+	// import/export is backwards compatible, and we must write code specific to the verison we're dealing with.
+	if dep.Version != 3 {
+		panic("expected deployment version 3")
+	}
+	var state apitype.DeploymentV3
+	_ = json.Unmarshal(dep.Deployment, &state)
+
+	// ... perform edits on the state ...
+
+	// marshal out updated deployment state
+	bytes, _ := json.Marshal(state)
+	dep.Deployment = bytes
+	// import our edited deployment state back to our stack
+	_ = stack.Import(ctx, dep)
+}
+
+func ExampleStack_Import() {
+	ctx := context.Background()
+	stackName := FullyQualifiedStackName("org", "project", "stack")
+	stack, _ := SelectStackLocalSource(ctx, stackName, filepath.Join(".", "program"))
+	dep, _ := stack.Export(ctx)
+	// import/export is backwards compatible, and we must write code specific to the verison we're dealing with.
+	if dep.Version != 3 {
+		panic("expected deployment version 3")
+	}
+	var state apitype.DeploymentV3
+	_ = json.Unmarshal(dep.Deployment, &state)
+
+	// ... perform edits on the state ...
+
+	// marshal out updated deployment state
+	bytes, _ := json.Marshal(state)
+	dep.Deployment = bytes
+	// import our edited deployment state back to our stack
+	_ = stack.Import(ctx, dep)
+}
+
```

Comment:
We should consider adding an example of working _through_ `pending_operations` e.g.

- export state
- delete pending resources from the state
- import updated state object
- instruct the user to manually clean up any leaked provider resources

Reply:
A simple export/import does clear pending operations according to our docs: https://www.pulumi.com/docs/troubleshooting/#interrupted-update-recovery

I do agree that our story here is weak. My goal here is to just expose the building blocks. We know we'll be investing in the product to make this experience better (and the current cancel/import/export workflow is not the answer).